graphqlapi/src/main/java/io/stargate/graphql/schema/graphqlfirst/processor/SchemaProcessor.java

```diff
@@ -289,13 +290,19 @@ private TypeDefinitionRegistry parse(String source) throws GraphqlErrorException
       return parser.parse(source);
     } catch (SchemaProblem schemaProblem) {
       List<GraphQLError> schemaErrors = schemaProblem.getErrors();
+      // Convert to JSON explicitly, because the parser sometimes returns errors that also implement
+      // java.lang.Exception (e.g. NonSDLDefinitionError), and those get formatted with a full stack
+      // trace by default.
+      List<Map<String, Object>> errorsJson =
+          schemaErrors.stream().map(GraphQLError::toSpecification).collect(Collectors.toList());
```

Comment:
I ran into this by chance while checking something else. A simple reproducing case:
```graphql
mutation {
  deploySchema(
    keyspace: "test"
    schema: """
    type Foo { id ID }
    """
  ) {
    version
  }
}
```

Reply:
maybe it's worth adding an integration test for it?

Reply:
Done in 4b558c98